AuthPass - Password Manager

hpoul opened the pull request:
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

Hello,
I am the developer of AuthPass, a cross platform password manager. See https://authpass.app/
I don't have much experience with linux development in specific though. So hopefully the packaging worked out correct.

thanks,
  Herbert

hfiguiere commented on .github/workflows/linux_flatpak.yaml:
```diff
@@ -0,0 +1,27 @@
+on:
```
you don't need the github actions here.

hpoul replied:
it is/was the easiest to test - both the building and updating.. if you don't mind i'd leave them in if I need to do some testing or building it manually.. they are configured to only build on specific branches.

hfiguiere commented on .github/workflows/update.yaml:
```diff
@@ -0,0 +1,35 @@
+name: Check for updates
```
f-e-d-c will work out of the box, you don't need this.

hfiguiere reviewed (changes requested):
Since you are upstream, the icons, .desktop and metainfo.xml files shouldl be managed upstream. (neither is specific to flatpak, and the lack of icon and desktop file is an actual bug for a  linux desktop app)

hfiguiere commented:
bot, build app.authpass.AuthPass

hfiguiere commented:
```
builddir/files/share/appdata/app.authpass.AuthPass.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• tag-missing           : <url> is not present
Validation of files failed
```

hpoul commented on app.authpass.AuthPass.yaml:
```diff
@@ -0,0 +1,61 @@
+      - install -Dpm644 app.authpass.AuthPass.metainfo.xml /app/share/metainfo/app.authpass.AuthPass.metainfo.xml
+    sources:
+      - type: file
+        url: https://data.authpass.app/data/artifacts/authpass-linux-1.9.6_1904.tar.gz
```
I think it would actually be possible by building on an aarch64 host, but cross compilation is still not possible 🤔 https://github.com/flutter/flutter/issues/74929

I've added the `flathub.json`

hpoul commented on app.authpass.AuthPass.yaml:
```diff
@@ -0,0 +1,61 @@
+app-id: app.authpass.AuthPass
+runtime: org.gnome.Platform
```
I don't think so, changed to freedesktop.

hpoul commented on app.authpass.AuthPass.metainfo.xml:
```diff
@@ -0,0 +1,56 @@
+        </screenshots>
+        <releases>
+    <release version="1.9.6_1904" date="2022-06-10"/>
+  </releases>
```
added `url`s and `content_rating`.

hpoul commented:
Thanks for the comments @hfiguiere 

* I've now moved the metadata into the upstream git repo and checkout from that repo. Right now those files are not included in the `.tar.gz`..
* Only build x86_64
* added `url` and `content-rating`
* switched to using freedesktop instead of gnome

Regarding github actions: They now only build on specific branches to make it easier for me to test the build process. If that's still a problem let me know and i'll delete them.

hfiguiere commented:
You can also trigger a build like that:

bot, build app.authpass.AuthPass

hpoul commented:
ok, i thought only admins could do that.. i've removed the workflows now @hfiguiere

x80486 commented on app.authpass.AuthPass.yaml:
```diff
@@ -0,0 +1,61 @@
+command: authpass
+finish-args:
+  # Filesystem and network access
+  - --filesystem=home
```
I think while people can really use any location(s) for the databases, constraining this a little bit more to `Documents` and/or `Downloads` might be better.

hfiguiere replied:
It's a Flutter app so I don't really have hopes for it to support XDG properly.

But maybe it needs a `persist` for the settings directory.

hpoul replied:
right now it stores 
* app settings into `$HOME/.authpass/`
* Cached files from cloud storage into `$TMPDIR`
* "internal" password files go into `$XDG_DATA_HOME` by default. But the user can obviously open a file at any location using a native open file dialog.

JakobDev replied:
> app settings into $HOME/.authpass/

settings don't belong tehere, even outside a Flatpak. Please use $XDG_CONFIG_HOME.

> Cached files from cloud storage into $TMPDIR

Note that $TMPDIR can be removed after a Flatpak is closed. If you want to use the cache longer than the App runs, use $XDG_CACHE_HOME. Otherwise $TMPDIR is the right choice.

hpoul replied:
@JakobDev okay I've implemented the changes.. https://github.com/authpass/authpass/commit/be25e8d05a8e3a18e56d21522118c7f892feae74#diff-f966a0c220ffee66c7b945faa18a71acd3f323788477a7ca483721e33572e171R59-R62

but it would be nice if we could still get this finished before I create a new release.

hfiguiere commented on app.authpass.AuthPass.yaml:
```diff
@@ -0,0 +1,61 @@
+  - --device=dri
+
+cleanup:
+  - /include
```
no `*.a` left behind?

hpoul replied:
Not sure if I fully understand that `cleanup` property.. but i guess the only thing which could leave anything behind is the `keybinder` build.. i've searched through the repo and found e.g. https://github.com/flathub/io.github.Pithos/blob/master/io.github.Pithos.json which uses

```
"cleanup": ["/lib/*.la", "/include", "/share", "/lib/pkgconfig"]
```

(although [others don't use any cleanup at all](https://github.com/flathub/com.github.KRTirtho.Spotube/blob/master/com.github.KRTirtho.Spotube.yml) 🤷‍♂️)

hfiguiere commented on app.authpass.AuthPass.yaml:
```diff
@@ -0,0 +1,61 @@
+      - install -Dpm644 authpass-git/metadata/linux/app.authpass.AuthPass.metainfo.xml
+        /app/share/metainfo/app.authpass.AuthPass.metainfo.xml
+    sources:
+      - type: file
```
if you made this 
```suggestion
      - type: archive
        dest: authpass
```

You wouldn't need to extract the archive.

hpoul commented:
bot, build app.authpass.AuthPass

hpoul commented:
bot, build app.authpass.AuthPass

barthalion commented:
```
~ 4.7s ❱ flatpak run app.authpass.AuthPass
authpass: error while loading shared libraries: libsecret-1.so.0: cannot open shared object file: No such file or directory
```

You can add https://github.com/flathub/shared-modules/tree/master/libsecret as git submodule and reference it from your manifest.

hpoul commented:
bot, build app.authpass.AuthPass

hpoul commented:
bot, build app.authpass.AuthPass

hpoul commented:
> You can add https://github.com/flathub/shared-modules/tree/master/libsecret as git submodule and reference it from your manifest.

thanks for the pointer.. first i tried switching back to the gnome runtime.. but I guess only including libsecret plus gtk might be the better option 🤔 
maybe give the latest build another try 

> ```
> flatpak install --user https://dl.flathub.org/build-repo/95565/app.authpass.AuthPass.flatpakref
> ```

barthalion commented:
Yup, it works now.

/merge
